lib/RingBuffer.js

```diff
@@ -0,0 +1,88 @@
+var util = require('util');
+var EventEmitter = require('events').EventEmitter;
+
+/**
+ * RingBuffer is a Writable Stream that just stores the last N records in
+ * memory.
+ *
+ * @param {Object} [options] - with the following fields:
+ * @param {Object.limit} [options.limit] - number of records to keep in memory
+ */
+var RingBuffer = module.exports = function RingBuffer(options) {
+    this.limit = options && options.limit ? options.limit : 100;
+    this.writable = true;
+    this.records = new Array(this.limit);
+    this._idx = -1;
+    EventEmitter.call(this);
+};
+
+util.inherits(RingBuffer, EventEmitter);
+
+RingBuffer.prototype.write = function (record) {
```

Comment:
You'll definitely want to name your functions here, to make it easier to find them in core dumps.

e.g.

```js
RingBuffer.prototype.write = function write (record) {
...
```